webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { updateSelected } from './TypeAheadSelectActions';
+import { selectOptions, selectSelected } from './TypeAheadSelectSelectors';
+import reducer from './TypeAheadSelectReducer';
+import TypeAheadSelect from './TypeAheadSelect';
+
+const ConnectedTypeAheadSelect = ({
+  id,
+  options = [],
+  selected,
+  defaultSelected,
+  allowNew,
+  multiple,
+}) => {
+  const _selected = useSelector(state => selectSelected(state, id, selected));
+  const _options = useSelector(state => selectOptions(state, id, options));
```

Comment:
It is difficult to keep track on props that came from server or from redux, and may lead to some bugs.

Imo the way to go is to create some initialUpdate action or retrieve all the props by an API call.